azurerm/resource_arm_recovery_services_network_mapping.go

```diff
@@ -0,0 +1,194 @@
+package azurerm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/recoveryservices/mgmt/2018-01-10/siterecovery"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmRecoveryServicesNetworkMapping() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmRecoveryNetworkMappingCreate,
+		Read:   resourceArmRecoveryNetworkMappingRead,
+		Update: nil,
+		Delete: resourceArmRecoveryNetworkMappingDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"recovery_vault_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"source_recovery_fabric_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"target_recovery_fabric_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"source_network_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+			"source_network_id": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateFunc:     validate.NoEmptyStrings,
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
Yes, the API returns another casing of the  "name" part of the ids, Thats also why I had to update line 58 of azurerm/helpers/azure/resourceid.go

Reply:
Now that we are fixing this at the d.Set(id) level can we remove this?

Reply:
Unfortunately no. The Set(id) fix handles the casing of the static word Subscription. But if I use uppercase letters in the name of my network it will come back in all lowercase.

We could lowercase the entire id before saving it, but that does not feel right.